ksql-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -137,10 +138,15 @@ private void validateQuery(final QueryMetadata query, final ConfiguredStatement<
       final SchemaRegistryClient srClient = serviceContext.getSchemaRegistryClient();
 
       if (!AvroUtil.isValidSchemaEvolution(persistentQuery, srClient)) {
+        final org.apache.avro.Schema avroSchema = SchemaUtil.buildAvroSchema(
+            persistentQuery.getPhysicalSchema().valueSchema(),
+            persistentQuery.getSinkName()
+        );
         throw new KsqlStatementException(String.format(
-            "Cannot register avro schema for %s as the schema registry rejected it, "
-                + "(maybe schema evolution issues?)",
-            persistentQuery.getResultTopic().getKafkaTopicName()),
+            "Cannot register avro schema for %s as the schema is incompatible.%n"
+                + "schema: %s",
```

Comment:
This exception is definitely better, as it includes the schema KSQL is trying to use.  What would be better still would be to include the schema in the schema registry in the message e.g.

`
Can not register avro schema for topic X as the schema is incompatible with the current schema version registered for the topic.
KSQL schema: {...}
Registered schema: {}
`

If you move the exception into the method, as discussed above, then in `testCompatibility` returns false you could grab the latest version of the schema using the following and include it in the exception as well:

```
// Compute this once:
final String subject = topicName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX

// Grab latest schema using:
final String registeredSchema = schemaRegistryClient
          .getLatestSchemaMetadata(subject)
          .getSchema()
```